closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -1706,6 +1725,32 @@ private static void checkProviderFormat(String providerInfo, boolean update) {
                 printStackTraceAndExit("Provider format is incorrect: " + providerInfo);
             }
         }
+
+        private void loadProfileCheck(String profileID) {
+            Enumeration<?> pNames = securityProps.propertyNames();
+            while (pNames.hasMoreElements()) {
+                String name = (String) pNames.nextElement();
+                if (name.startsWith(profileID + '.') && name.contains(".jce.provider.")) {
+                    profileCheckProviderNames.add(name);
+                } else if (name.startsWith(profileID + '.')) {
+                    profileCheckPropertyNames.add(name);
+                }
+            }
+        }
+
+        private void checkProfileCheck(String profileID) {
+            if (!profileCheckProviderNames.isEmpty()) {
+                printStackTraceAndExit(
+                        "The order of providers in profile " + profileID
```

Comment:
This message may be misleading; the _order_ may be correct, but there may be a problem with the keys (bad starting key, missing or repeated keys).

Reply:
For the provider name check, it only check if the order numbers of providers are consecutive or not. For others, like bad starting key, missing key, etc. For example, 

`RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.provider.1 `wrong typo to `RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.prOOvider.1,` the “`if (!profileCheckPropertyNames.isEmpty()`)” will detect this, since this key can not be recognized and parsed in restricted security mode.